base/iterators.jl

```diff
@@ -71,16 +71,20 @@ and_iteratoreltype(a, b) = EltypeUnknown()
     Iterators.reverse(itr)
 
 Given an iterator `itr`, then `reverse(itr)` is an iterator over the
-same collection but in the reverse order.
+same collection but in the reverse order.   (By default, this returns
+an `Iterators.Reverse` object wrapping `itr`, which is iterable
+if the corresponding [`iterate`](@ref) methods are defined, but some `itr` types
+may implement more specialized `Iterators.reverse` behaviors.)
```

Comment:
Can I suggest moving this down? 

Then first things are: "It reverses. It's lazy, see reverse. Some types may not work." in that order. Then, after that, start to mention internal details like iterate and types.

Reply:
@stevengj Thoughts on this?

Reply:
re-ordered the text as suggested.